modules/uid2IdSystem_shared.js

```diff
@@ -5,13 +5,99 @@ function isValidIdentity(identity) {
   return !!(typeof identity === 'object' && identity !== null && identity.advertising_token && identity.identity_expires && identity.refresh_from && identity.refresh_token && identity.refresh_expires);
 }
 
+function isNormalizedPhone(phone) {
+  return /^\+[0-9]{10,15}$/.test(phone);
+}
+
+function isBase64Hash(value) {
+  if (!(value && value.length === 44)) {
+    return false;
+  }
+
+  try {
+    return btoa(atob(value)) === value;
+  } catch (err) {
+    return false;
+  }
+}
+
+function isCSTGOptionsAndGetIdentity(
```

Comment:
This function seems to be doing too many things, and the name is confusing too.

Reply:
I will split it into one for checking  serverPublicKey and subscriptionId is valid in opts and the other for extracting identity,